catalog/to-consul/consul_node_services_client_test.go

```diff
@@ -0,0 +1,168 @@
+package catalog
+
+import (
+	"testing"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPreConsulOnePointSevenClient_NodeServicesWithTag(t *testing.T) {
+	t.Parallel()
+	cases := map[string]struct {
+		ConsulServices []api.CatalogRegistration
+		Exp            []ConsulService
+	}{
+		"no services": {
+			ConsulServices: nil,
+			Exp:            nil,
+		},
+		"no services on k8s node": {
+			ConsulServices: []api.CatalogRegistration{
+				{
+					Node:    "not-k8s",
+					Address: "127.0.0.1",
+					Service: &api.AgentService{
+						ID:      "svc-id",
+						Service: "svc",
+					},
+				},
+			},
+			Exp: nil,
+		},
+		"service with k8s tag on different node": {
+			ConsulServices: []api.CatalogRegistration{
+				{
+					Node:    "not-k8s",
+					Address: "127.0.0.1",
+					Service: &api.AgentService{
+						ID:      "svc-id",
+						Service: "svc",
+						Tags:    []string{"k8s"},
+					},
+				},
+			},
+			Exp: nil,
+		},
+		"service on k8s node but without tag": {
```

Comment:
Similar to the ent tests, it would be good to duplicate this test but put non-`k8s` tags on it.